src/main/java/gift/service/KakaoOAuthService.java

```diff
@@ -0,0 +1,93 @@
+package gift.service;
+
+import gift.dto.api.KakaoTokenResponseDto;
+import gift.dto.api.KakaoUserResponseDto;
+import gift.exception.KakaoOAuthException;
+import io.netty.channel.ChannelOption;
+import java.time.Duration;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.http.MediaType;
+import org.springframework.http.client.reactive.ReactorClientHttpConnector;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.BodyInserters;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+import reactor.netty.http.client.HttpClient;
+
+@Service
+public class KakaoOAuthService {
+
+    @Value("${kakao.clientId}")
+    private final String clientId;
+
+    @Value("${kakao.auth-url}")
+    private final String authUrl;
+
+    @Value("${kakao.api-url}")
+    private final String apiUrl;
+
+    @Value("${kakao.redirect-uri:}")
```

Comment:
```suggestion
    @Value("${kakao.redirect-uri}")
```
이건 오타일까요? 디폴트 값 지정하려고 했던걸까요? 디폴트 값 지정하려면 아래처럼 할 수도 있습니다😊
```suggestion
    @Value("${kakao.redirect-uri:localhost:8080}")
```

Reply:
수정 제안 감사합니다! 😊
맞습니다 `:` 을 붙여 기본값을 줄 생각이었는데, 값까지 채워 넣는 걸 깜빡했습니다.

다시 생각해보니 redirect-uri 는 환경마다 반드시 값을 지정해야 하는 필수 설정이기 때문에 기본값 없이 두는 편이 더 안전할 것 같아서 기본값을 지정하지 않는 쪽으로 수정하도록 하겠습니다!